### PR TITLE
OV-59: add video api

### PR DESCRIPTION
**Pull request by Sanchousina**
- Created migration file that creates video table
- Video Repository
- Video Service
- Video Controller
- Implemented video CRUD operations

image
image
image
image
image

### CODE REVIEW

File: backend/src/bundles/videos/video.controller.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            handler: (options) =>
+                this.delete(
+                    options as ApiHandlerOptions<{
+                        params: VideoGetOneRequestDto;
+                    }>,
+                ),
+        });
+    }
+
+    private async findAll(): Promise<ApiHandlerResponse> {
```
Comment (o-nedashkivska) [inline backend/src/bundles/videos/video.controller.ts:88]: Please, add swagger docs for all handlers

File: shared/src/bundles/videos/types/update-video-request-dto.type.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { type CreateVideoRequestDto } from './create-video-request-dto.type.js';
+
+type UpdateVideoRequestDto = Partial<CreateVideoRequestDto>;
```
Comment (o-nedashkivska) [inline shared/src/bundles/videos/types/update-video-request-dto.type.ts:3]: How are we going to identify video: by its id or by userId and video name? If second option then userId and name fields are required.

Comment (o-nedashkivska) [reply shared/src/bundles/videos/types/update-video-request-dto.type.ts:3]: Sorry, I didn't see usage examples at first. Now I see that to delete a video we will use its id.
But in what cases we are going to change userId or url for existing video?

File: backend/src/bundles/videos/video.controller.ts (hunk 1 above)
Comment (Sanchousina) [reply backend/src/bundles/videos/video.controller.ts:88]: Thanks for reminding me, I forgot

File: shared/src/bundles/videos/types/update-video-request-dto.type.ts (hunk 2 above)
Comment (Sanchousina) [reply shared/src/bundles/videos/types/update-video-request-dto.type.ts:3]: For now I just implemented basic update for video and don't know about cases. We can refine it in the future, when we'll know what we will update, maybe?

File: backend/src/common/types/repository.type.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
 type Repository<T = unknown> = {
-    find(): Promise<T>;
+    find(id: string): Promise<T | null>;
     findAll(): Promise<T[]>;
     create(payload: unknown): Promise<T>;
-    update(): Promise<T>;
-    delete(): Promise<boolean>;
+    update(id: string, payload: unknown): Promise<T>;
```
[the remark this answers is not in the document]
Comment (Sanchousina) [reply backend/src/common/types/repository.type.ts:5]: What do you mean by the same for services? In the service, the update method returns just T, is it wrong? I think I don't need to return T | null in service

[the remark this answers is not in the document]
Comment (sofiia-trokhymchuk) [reply backend/src/common/types/repository.type.ts:5]: Oh yes, sorry, you handle cases where entity doesn't exist. Good job! But don't forget to change `update` return type in repository type defintion

File: shared/src/bundles/videos/validation-schemas/create-video.validation-schema.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+type CreateVideoValidationDto = {
+    userId: z.ZodString;
+    name: z.ZodString;
+    url: z.ZodString;
+};
+
+const createVideo = z
+    .object<CreateVideoValidationDto>({
+        userId: z
+            .string({ required_error: VideoValidationMessage.USER_ID_REQUIRE })
```
[the remark this answers is not in the document]
Comment (Sanchousina) [reply shared/src/bundles/videos/validation-schemas/create-video.validation-schema.ts:14]: But it is triggered by `.uuid()` check, that's why I didn't use `.min(1)`. Should I still apply min(1) check?

image

[the remark this answers is not in the document]
Comment (sofiia-trokhymchuk) [reply shared/src/bundles/videos/validation-schemas/create-video.validation-schema.ts]: If it is a desired validation behavior then fine)

File: shared/src/bundles/videos/enums/videos-api-path.enum.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+const VideosApiPath = {
+    ROOT: '/',
+    VIDEO: '/:videoId',
```
[the remark this answers is not in the document]
Comment (Sanchousina) [reply shared/src/bundles/videos/enums/videos-api-path.enum.ts]: Done

File: shared/src/bundles/videos/enums/video-validation-message.enum.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const VideoValidationMessage = {
+    VIDEO_DOESNT_EXIST: 'Video with this id does not exist',
+    USER_ID_REQUIRE: 'User id is required',
+    UUID_WRONG: 'Invalid UUID',
+    NAME_REQUIRE: 'The video name is required',
+    NAME_LENGTH: 'The video name should have at least 1 character',
+    URL_REQUIRE: 'The video url is required',
```
[the remark this answers is not in the document]
Comment (Sanchousina) [reply shared/src/bundles/videos/enums/video-validation-message.enum.ts]: This and others done

File: shared/src/bundles/videos/enums/videos-api-path.enum.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const VideosApiPath = {
+    ROOT: '/',
+    VIDEO: '/:id',
```
Comment (sofiia-trokhymchuk) [inline shared/src/bundles/videos/enums/videos-api-path.enum.ts:3]:
```suggestion
    ID: '/:id',
```

**Review by sofiia-trokhymchuk — CHANGES_REQUESTED** [review]

**Review by sofiia-trokhymchuk — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**